### PR TITLE
Optimize function `mapUpdate`

### PR DESCRIPTION
**Pull request by CurtizJ**
### Changelog category (leave one):
- Performance Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Optimize function `mapUpdate`.

### CODE REVIEW

File: tests/queries/0_stateless/02169_map_functions.sql [hunk 1]
```diff
@@ -15,6 +15,11 @@ SELECT mapApply((x, y) -> (x, x + 1), materialize(map(1, 0, 2, 0)));
 SELECT mapApply((x, y) -> ('x', 'y'), map(1, 0, 2, 0));
 SELECT mapApply((x, y) -> ('x', 'y'), materialize(map(1, 0, 2, 0)));
 
+SELECT mapUpdate(map('k1', 1, 'k2', 2), map('k1', 11, 'k2', 22));
+SELECT mapUpdate(materialize(map('k1', 1, 'k2', 2)), map('k1', 11, 'k2', 22));
+SELECT mapUpdate(map('k1', 1, 'k2', 2), materialize(map('k1', 11, 'k2', 22)));
+SELECT mapUpdate(materialize(map('k1', 1, 'k2', 2)), materialize(map('k1', 11, 'k2', 22)));
```
Comment (yakov-olkhovskiy) [inline tests/queries/0_stateless/02169_map_functions.sql:21]: would be nice to cover partially overlaping sets - like mapUpdate(map('k1', 1, 'k2', 2, 'k3', 3), map('k2', 22, 'k3', 33, 'k4', 44))

**Review by yakov-olkhovskiy — APPROVED** [review]
LGTM, see comment on test

File: tests/queries/0_stateless/02169_map_functions.sql (hunk 1 above)
Comment (CurtizJ) [reply tests/queries/0_stateless/02169_map_functions.sql:21]: There is such test, but I agree that it's better to add more tests. I'll add some.

**Comment by CurtizJ** [comment]
Perf tests are ok:
Screenshot 2023-03-30 at 00 17 39

**Pull request merged**